keyvi/include/keyvi/dictionary/fsa/internal/ivalue_store.h

```diff
@@ -50,18 +47,25 @@ namespace internal {
  * Do not forget to add the new value store to ValueStoreFactory
  */
 enum value_store_t {
-  NULL_VALUE_STORE = 1,               //!< NullValueStore
-  INT_VALUE_STORE = 2,                //!< IntValueStore
-  STRING_VALUE_STORE = 3,             //!< StringValueStore
-  JSON_VALUE_STORE_DEPRECATED = 4,    // !< JsonValueStoreDeprecated
-  JSON_VALUE_STORE = 5,               // !< JsonValueStore
-  INT_INNER_WEIGHTS_VALUE_STORE = 6,  // !< IntInnerWeightsValueStore
+  KEY_ONLY = 1,          //!< NullValueStore
```

Comment:
Maybe this refactoring is a good time to use scoped enums here ?

Reply:
Excellent Idea!